src/Task.php

```diff
@@ -200,11 +200,15 @@ public function autoCleanup()
                     ->min('id');
                 self::results()
                     ->where('id', '<', $oldest_id)
-                    ->delete();
+                    ->chunkById(500, function ($results) {
+                        $results->delete();
+                    });
             } else {
                 self::results()
                     ->where('ran_at', '<', Carbon::now()->subDays($this->auto_cleanup_num - 1))
-                    ->delete();
+                    ->chunkById(500, function ($results) {
+                        $results->delete();
+                    });
```

Comment:
You're calling `delete` on a collection, what you want is a higher order function:
```suggestion
                    ->chunkById(500, function ($results) {
                        $results->each->delete();
                    });
```

Reply:
Ah dang, I was hoping to avoid the individual calls to delete and that it'd behave like calling the function on the Builder like this did previously

Reply:
Another option:
```
do {
    $rowsDeleted = self::results()
        ->where('ran_at', '<', Carbon::now()->subDays($this->auto_cleanup_num - 1))
        ->limit(500)
        ->getQuery()
        ->delete();
} while ($rowsDeleted > 0);
```